nstar/src/main.rs

```diff
@@ -12,307 +12,110 @@
-            time::Duration::from_secs(2),
-            api::client::Client::new(&host),
-        )
-        .await
-        {
-            Ok(Ok(client)) => client,
-            Ok(Err(e)) => {
-                writeln!(terminal, "Failed to connect: {:?}", e)?;
-                if interactive {
-                    time::sleep(time::Duration::from_secs(1)).await;
-                    continue 'outer;
-                } else {
-                    break;
-                }
-            }
-            Err(_) => {
-                if interactive {
-                    writeln!(terminal, "Failed to connect: timeout")?;
-                    time::sleep(time::Duration::from_secs(1)).await;
-                    continue 'outer;
-                } else {
-                    break;
-                }
-            }
-        };
-
-        writeln!(terminal, "Connected to {}", opt.host)?;
+        writeln!(terminal, "Connected to {}", &opt.host)?;
 
         loop {
             select! {
                 notification = client.next() => {
-                    if interactive {
-                        if let Some(Ok(n)) = notification {
-                            pretty::notification(&mut terminal, &n);
-                        } else {
-                            break;
-                        }
+                    if let Some(Ok(n)) = notification {
+                        pretty::notification(&mut terminal, &n);
+                    } else {
+                        break;
                     }
                 }
                 input = input.next() => {
-                    match input.as_deref() {
-                        Some("quit") | Some("exit") => break 'outer,
-                        Some(ref input) => {
-                            match process(&mut client, &mut terminal, &input).await {
-                                Ok(Some(msg)) => writeln!(&mut terminal, "⚠ {}", msg)?,
-                                Ok(None) => writeln!(&mut terminal, "✓ {}", input)?,
-                                Err(e) => {
-                                    writeln!(&mut terminal, "⚠ {:?}", e)?;
-                                    break;
-                                }
-
-                            }
-                        }
-                        None => break 'outer,
-                    }
+                    let input: &str = if input.is_some() {
+                        input.as_deref().unwrap()
+                    } else {
+                        break 'outer;
+                    };
+                    match parse_prompt(input) {
+                        Ok(PromptCommand::Prompt(cmd)) => match cmd {
+                            Prompt::Help => print_help(&mut terminal)?,
+                            Prompt::Quit => break 'outer,
+                        },
+                        Ok(PromptCommand::Northstar(cmd)) => {
+                            let response = send_request(&mut client, &cmd).await?;
```

Comment:
this will try to send something but nstar will exit if there is no connection...
we should try a reconnect before failing to send the request

Reply:
I think you are right! we should handle this error in the particular case of a disconnection :thinking: